Source/Diagnostics/Diagnostics.cpp

```diff
@@ -362,6 +361,14 @@ Diagnostics::FilterComputePackFlush (int step, bool force_flush)
     WARPX_PROFILE("Diagnostics::FilterComputePackFlush()");
     MovingWindowAndGalileanDomainShift (step);
 
+    const auto& warpx = WarpX::GetInstance();
+    const auto finest_level = warpx.finestLevel();
+
+    for (auto& buf : m_mf_output){
+        if(buf.size() > finest_level+1) buf.pop_back();
+    }
```

Comment:
I would suggest a `resize` here. Moreover, I think that adding a comment before the `for` loop could help developers working on this part of the code in the future.

Reply:
Agreed.